CIPs/cip-0032.md

```diff
@@ -0,0 +1,120 @@
+
+### Uptime Score
+
+One proposal was to use the uptime score as the main incentive mechanism. In this model, a long outage in the Attestation Service would also have weight on a Validator's uptime score. The downside of this approach is that it would require a hardfork to implement. Given that this is just the first stop on a long iteration process, we decided to go with something adjustable with a higher turnaround process.
+
+### Using completion rate
+
+Instead of using a min-completion count, we could use the total rate of completion to reward or punish a validator. This was decided against until we have strong confidence in the completion rate metric. It's possible for a simple metric to be gamed by malicious actors. We also want the metric to reflect the difference between a user not having received the SMS vs quitting the flow prematurely. We also need more users onboarding before we can build confidence in this metric. We view the min-completion count as the first step in this direction.
+
+### Foundation votes
+
+The Foundation holds a large amount of Celo and uses it [today to incentivize behavior to support a healthy network](https://docs.celo.org/validator-guide/celo-foundation-voting-policy). The Foundation voting does already check that validators are running a highly available attestation service; however we felt that there should be an additional incentive for the following reasons:
+
+1. Some validators are not eligible for foundation votes. We need the solution to apply to all validators to improve the overall AS availability.
+2. Voting frequency is every 3 months, which is not immediate enough to motivate an operator to regularly monitor and maintain their AS availability.
+
+### Governance Slashing
+
+We have a generic slashing contract which allows anyone to slash for any reason via a governance proposal. This process is good for one-off slashing events, but requires a large amount of coordination for each event. We wanted to build something more automatic where the rules are stated explicitly ahead of time.
+
+### Signal online
+
+We considered introducing a mechanism for the AS to signal that it was still online in the case that it never had the opportunity to be selected for an attestation. The goal of this mechanism was to add more assurance that a validator wouldn't be incorrectly slashed if they didn't have a chance to naturally prove that they were online. Instead we're mitigating this risk in the following ways:
+
+1. Starting with a conservative slashing penalty.
+2. Requiring a single attestation in a time period where we'd naturally expect 10 to have occurred.
+3. Slashing can only occur if the AS was selected for at least one attestation in the time period.
```

Comment:
This feels like it contradicts (2), why "one" instead of "ten"?

Reply:
2 is saying that if that the contract's min-bar is very low. Based on a snapshot of traffic in November, there should be 10 attestations randomly assigned to an AS in the time-frame that the slasher only requires one.

3 is saying that in the case that there happens to be zero attestations requested from an AS in the time-frame, then there's a safety mechanism that will prevent the AS from being slashed. In other words, the contract won't incorrectly slash an AS just because there wasn't any traffic.